osier/models/dispatch.py

```diff
@@ -40,26 +49,46 @@ class DispatchModel():
     .. math::
         x_{u,t} \\leq \\textbf{CAP}_{u} \\ \\forall \\ u,t \\in U,T
 
+    3. Technologies may not exceed their ramp up rate.
+
+    .. math::
+        \\frac{x_{r,t} - x_{r,t-1}}{\\Delta t} = \\Delta P_{r,t} \\leq
+        (\\text{ramp up})\\textbf{CAP}_u\\Delta t \\ \\forall \\ r,t
+        \\in R \\subset U, T
+
+    or ramp down rate
+
+    .. math::
+        \\frac{x_{r,t} - x_{r,t-1}}{\\Delta t} = \\Delta P_{r,t} \\leq
+        -(\\text{ramp down})\\textbf{CAP}_u\\Delta t \\ \\forall \\ r,t
+        \\in R \\subset U, T
 
     Parameters
     ----------
     technology_list : list of :class:`osier.Technology`
         The list of :class:`Technology` objects to dispatch -- i.e. decide
         how much energy each technology should produce.
-    net_demand : List, :class:`numpy.ndarray`, :class:`unyt.array.unyt_array`
+    net_demand : list, :class:`numpy.ndarray`, :class:`unyt.array.unyt_array`, :class:`pandas.DataFrame`.
         The remaining energy demand to be fulfilled by the technologies in
-        :attr:`technology_list`. For example:
-
-        >>> from osier import DispatchModel
-        >>> from osier import Technology
-        >>> wind = Technology("wind")
-        >>> nuclear = Technology("nuclear")
-        >>> import numpy as np
-        >>> hours = np.arange(24)
-        >>> demand = np.cos(hours*np.pi/180)
-        >>> model = DispatchModel([wind, nuclear], demand)
-        >>> model.solve()
-
+        :attr:`technology_list`. The `values` of an object passed as
+        `net_demand` are used to create a supply constraint. See
+        :attr:`oversupply` and :attr:`undersupply`.
+        If a :class:`pandas.DataFrame` is passed, :mod:`osier` will try
+        inferring a `time_delta` from the dataframe index. Otherwise, the
+        :attr:`time_delta` must be passed or the default is used.
+    time_delta : str, :class:`unyt.unyt_quantity`, float, int
+        Specifies the amount of time between two time slices. The default is
+        one hour. Can be overridden by specifying a unit with the value. For
+        example:
+
+        >>> time_delta = "5 minutes"
+        >>> from unyt import days
+        >>> time_delta = 30*days
+
+        would both work.
+    power_units : str, :class:`unyt.unyt_quantity`, float, int
```

Comment:
How can the power units be a float or integer?

Reply:
Cannot be! I copied the string from time_delta